jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipResponse.java

```diff
@@ -38,13 +39,20 @@
 public class GzipResponse extends Response.Wrapper
 {
     public static Logger LOG = LoggerFactory.getLogger(GzipResponse.class);
-    private static final byte[] GZIP_HEADER = new byte[]{(byte)0x1f, (byte)0x8b, Deflater.DEFLATED, 0, 0, 0, 0, 0, 0, 0};
+
+    // Per RFC-1952 this is the "unknown" OS value byte.
+    private static final byte OS_UNKNOWN = (byte)0xFF;
+    private static final byte[] GZIP_HEADER = new byte[]{
+        (byte)0x1f, (byte)0x8b, Deflater.DEFLATED, 0, 0, 0, 0, 0, 0, OS_UNKNOWN
+    };
+    // Per RFC-1952, the GZIP trailer is 8 bytes
+    public static final int GZIP_TRAILER_SIZE = 8;
 
     public static final HttpField VARY_ACCEPT_ENCODING = new PreEncodedHttpField(HttpHeader.VARY, HttpHeader.ACCEPT_ENCODING.asString());
 
     private enum GZState
     {
-        MIGHT_COMPRESS, NOT_COMPRESSING, COMMITTING, COMPRESSING, FINISHED
+        MIGHT_COMPRESS, NOT_COMPRESSING, COMMITTING, COMPRESSING, FINISHING, FINISHED
```

Comment:
Some javadoc against each state would be helpful (we should have had it before).